src/internal/types.ts

```diff
@@ -1,5 +1,9 @@
 import type { IsAny, IsNever } from "type-fest";
 
+declare const __brand: unique symbol;
+
+export type Brand<T> = { [__brand]: T };
```

Comment:
1. Should brand only accept symbols?
2. To make the API more fail-proof I'd limit the usage to also branding the object, so something like this:
```suggestion
export type Branded<T, Brand extends symbol> = T & { [__brand]: Brand };
```

Reply:
@eranhirsch 

1. The initial idea was to create a general-purpose utility type but since it's an internal type and we don't expect anything other than symbol I think it makes sense to constrain it.
2. I thought about it but ended up with just `Brand` type to have more flexibility. I will change it to what you proposed.

Reply:
Done